File: TestObjects/CSCompiler/Literals.cs
using System;
using System.Collections.Generic;
using System.Linq.Expressions;
using static ExpressionTreeTestObjects.Functions;
using static ExpressionTreeTestObjects.Categories;

namespace ExpressionTreeTestObjects {
    partial class CSCompiler {
        [TestObject(Literal)]
        internal static readonly Expression True = Expr(() => true);

        [TestObject(Literal)]
        internal static readonly Expression False = Expr(() => false);

        [TestObject(Literal)]
        internal static readonly Expression Nothing = Expr(() => (string?)null);

        [TestObject(Literal)]
        internal static readonly Expression Integer = Expr(() => 5);

        [TestObject(Literal)]
        internal static readonly Expression NonInteger = Expr(() => 7.32);

        [TestObject(Literal)]
        internal static readonly Expression String = Expr(() => "abcd");

        [TestObject(Literal)]
        internal static readonly Expression InterpolatedString = Expr(() => $"{new DateTime(2001, 3, 25)}");

        [TestObject(Literal)]
        internal static readonly Expression Type = Expr(() => typeof(string));
    }
}

File: TestObjects.VB/VBCompiler/Literals.vb
Partial Module VBCompiler
    <TestObject(Literal)>
    Friend [True] As Expression = Expr(Function() True)

    <TestObject(Literal)>
    Friend [False] As Expression = Expr(Function() False)

    <TestObject(Literal)>
    Friend NothingString As Expression = Expr(Function() CType(Nothing, String))

    <TestObject(Literal)>
    Friend [Nothing] As Expression = Expr(Function() Nothing)

    <TestObject(Literal)>
    Friend [Integer] As Expression = Expr(Function() 5)

    <TestObject(Literal)>
    Friend NonInteger As Expression = Expr(Function() 7.32)

    <TestObject(Literal)>
    Friend [String] As Expression = Expr(Function() "abcd")

    <TestObject(Literal)>
    Friend EscapedString As Expression = Expr(Function() """")

    <TestObject(Literal)>
    Friend ConstantNothingToObject As Expression = Expr(Function() Nothing)

    <TestObject(Literal)>
    Friend ConstantNothingToReferenceType As Expression = Expr(Of String)(Function() Nothing)

    <TestObject(Literal)>
    Friend ConstantNothingToValueType As Expression = Expr(Of Integer)(Function() Nothing)

    <TestObject(Literal)>
    Friend InterpolatedString As Expression = Expr(Function() $"{#2001-3-25#:yyyy-MM-dd}")
End Module